virttest/test_setup.py

```diff
@@ -678,9 +678,17 @@ def setup(self):
                       " for this test: %s", self.over_commit.proc_fs_value)
         LOG.debug("Number of large memory pages needed for this test: %s",
                   self.target_hugepages)
+
+        # Set hugepage may fail because of insufficient continual memory
+        # Compact memory to get more continual memory
+        with open("/proc/sys/vm/compact_memory", "w") as memory:
```

Comment:
@huangyum If we enable compact memory here this will become a default configuration. what do you think?
Maybe an interface should be added for control?

Reply:
I think it's ok to always compact memory before allocating hugepages. However, I guess it's better to be executed after drop caches in L688.

Reply:
> I think it's ok to always compact memory before allocating hugepages. However, I guess it's better to be executed after drop caches in L688.

Good point !